sdcm/nemesis.py

```diff
@@ -4960,6 +4962,82 @@ def are_gate_closed_messages_raised(log_reader):
+
+        def prepare_test_keyspace(node):
+            """ Create new table for verification missing data
+
+            Create new keyspace and table which allow to
+            verify that no operations come from banned node
+            """
+            with self.cluster.cql_connection_patient(node) as session:
+                session.execute(
+                    "CREATE KEYSPACE IF NOT EXISTS banned_keyspace \
+                        WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor' : 3};")
+                session.execute("CREATE TABLE IF NOT EXISTS banned_keyspace.table1 (id BIGINT PRIMARY KEY, name text);")
+
+        simulate_node_unavailability = node_operations.simulate_node_unavailability(use_iptables)
+        working_node = node_operations.get_random_free_verification_node(self.target_node)
+        target_host_id = self.target_node.host_id
+
+        prepare_test_keyspace(self.target_node)
+        with simulate_node_unavailability(self.target_node):
+            # target node stopped by Contextmanger. Wait while its status will be updated
+            wait_for(node_operations.check_node_seen_as_down_in_cluster, timeout=180, throw_exc=True,
+                     down_node=self.target_node)
+            self.log.debug("Remove node %s : hostid: %s with stopped scylla from cluster",
+                           self.target_node.name, target_host_id)
+            working_node.run_nodetool(f"removenode {target_host_id}", retry=3)
+            self.log.debug("Wait while all node update topology status")
+            wait_for(node_operations.check_node_removed_from_cluster, timeout=300, throw_exc=True,
+                     node=self.target_node)
+            # Run read barrier on working node to validate that it update raft topology state
+            self.cluster.read_barrier(working_node)
+
+        # Context manager at exit  start scylla on target node.
+        # But node already removed from cluster. So any operations from it
+        # should be banned. If query executed succesfull, raise an error
+        try:
+            with self.cluster.cql_connection_exclusive(node=self.target_node) as session:
+                stmt = SimpleStatement("INSERT INTO banned_keyspace.table1 (id, name) VALUES (1, 'name1');",
+                                       consistency_level=ConsistencyLevel.ALL)
+                session.execute(stmt)
+                raise BannedCQLOperation("wrong operations were executed")
+        finally:
+            # because target node removed from cluster, terminate it
+            # and bootstrap new one
+            LOGGER.debug("Terminating node %s", self.target_node.name)
+            self.cluster.terminate_node(self.target_node)
+            LOGGER.debug("Restore number of cluster nodes")
+            new_node = self._add_and_init_new_cluster_node()
+            LOGGER.debug("New node %s was added", new_node.name)
```

Comment:
Possibly not enough just to add, if RBNO isn't enabled, you need to run repair.
Otherwise, you may end up with possible data loss.

Reply:
Added repair for all nodes